### ISSUE TITLE
RabbitMQ connexion doesn't work (example MISP)

### ISSUE THREAD

**Issue by darcosion** [OpenCTI-Platform/connectors#54]
## Description

When I configure OpenCTI and wanting to add some connector, I've configured my connector, but they crash because the reach OpenCTI API, but cannot use correctly AQMP.

## Environment

1. Linux ubuntu-virtual-machine 5.3.0-40-generic #32~18.04.1-Ubuntu SMP Mon Feb 3 14:05:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux (with uname -a)
2. OpenCTI version: { e.g. OpenCTI 1.0.2 }
3. OpenCTI client: frontend 

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Install OpenCTI latest with docker
2. Configure it to a local use (localhost)
3. git clone https://github.com/OpenCTI-Platform/connectors
4. cd connectors/misp/src
5. sudo pip3 install -r requirements.txt
5. Configuring config.yml : 
```opencti:
  url: 'http://127.0.0.1:8080'
  token: 'myToken'

connector:
  id: 'myToken'
  type: 'EXTERNAL_IMPORT'
  name: 'MITRE ATT&CK'
  scope: 'identity,attack-pattern,course-of-action,intrusion-set,malware,tool,report'
  confidence_level: 3
  update_existing_data: True
  log_level: 'info'

mitre:
  enterprise_file_url: 'https://raw.githubusercontent.com/mitre/cti/master/enterprise-attack/enterprise-attack.json'
  pre_attack_file_url: 'https://raw.githubusercontent.com/mitre/cti/master/pre-attack/pre-attack.json'
  interval: 7 # Days
```

## Expected Output

Just wanting to see importation of files.

## Actual Output

```
$ python3 mitre.py 
INFO:root:Listing Threat-Actors with filters null.
INFO:root:Starting ping alive thread
test
INFO:root:Fetching MITRE datasets...
INFO:root:Connector has never run
INFO:root:Connector will run!
ERROR:pika.adapters.utils.selector_ioloop_adapter:Address resolution failed: gaierror(-2, 'Name or service not known')
ERROR:pika.adapters.utils.connection_workflow:getaddrinfo failed: gaierror(-2, 'Name or service not known').
ERROR:pika.adapters.utils.connection_workflow:AMQP connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - gaierror(-2, 'Name or service not known'); first exception - None.
ERROR:pika.adapters.utils.connection_workflow:AMQPConnectionWorkflow - reporting failure: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - gaierror(-2, 'Name or service not known'); first exception - None
ERROR:pika.adapters.blocking_connection:Connection workflow failed: AMQPConnectionWorkflowFailed: 1 exceptions in all; last exception - gaierror(-2, 'Name or service not known'); first exception - None
ERROR:pika.adapters.blocking_connection:Error in _create_connection().
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pika/adapters/blocking_connection.py", line 450, in _create_connection
    raise self._reap_last_connection_workflow_error(error)
  File "/usr/local/lib/python3.6/dist-packages/pika/adapters/utils/selector_ioloop_adapter.py", line 564, in _resolve
    self._flags)
  File "/usr/lib/python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known
```

 
## Additional information

When I see log of OpenCTI, I see : 
```
rabbitmq_1                               | 2020-03-05 13:04:01.574 [info] <0.16172.0> connection <0.16172.0> (172.19.0.13:52248 -> 172.19.0.2:5672): user 'guest' authenticated and granted access to vhost '/'
rabbitmq_1                               | 2020-03-05 13:04:01.589 [info] <0.16172.0> closing AMQP connection <0.16172.0> (172.19.0.13:52248 -> 172.19.0.2:5672, vhost: '/', user: 'guest')

```
 I gess the script can access to OpenCTI API, and to RabbitMQ, but crash after, maybe misconfiguration of pika ?

**Comment by richard-julien** [comment]
Hi @darcosion , can you copy/paste your docker compose configuration for this connector?
Thanks

**Comment by darcosion** [comment]
Related to https://github.com/OpenCTI-Platform/opencti/issues/547#issuecomment-595256134, I have two docker configurations, one for Grakn and one for the rest, there is my docker-compose.yml : 
```
compose.yml : 
    version: '3'
    services:
      grakn:
        image: graknlabs/grakn:1.6.2
        ports:
          - 48555:48555
        volumes:
          - grakndata:/grakn-core-all-linux/server/db
        restart: always
      redis:
        image: redis:5.0.5
        restart: always
      elasticsearch:
        image: docker.elastic.co/elasticsearch/elasticsearch:7.5.2
        volumes:
          - esdata:/usr/share/elasticsearch/data
        environment:
          - discovery.type=single-node
        restart: always
        ulimits:
          memlock:
            soft: -1
            hard: -1
          nofile:
            soft: 65536
            hard: 65536
        ports:
          - 9200:9200
          - 9300:9300
      minio:
        image: minio/minio:RELEASE.2019-10-12T01-39-57Z
        volumes:
          - s3data:/data
        ports:
          - "9000:9000"
        environment:
          MINIO_ACCESS_KEY: ${MINIO_ACCESS_KEY}
          MINIO_SECRET_KEY: ${MINIO_SECRET_KEY}
        command: server /data
        healthcheck:
          test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
          interval: 30s
          timeout: 20s
          retries: 3
      rabbitmq:
        image: rabbitmq:3.7.17-management
        environment:
          - RABBITMQ_DEFAULT_USER=${RABBITMQ_DEFAULT_USER}
          - RABBITMQ_DEFAULT_PASS=${RABBITMQ_DEFAULT_PASS}    
        restart: always
      opencti:
        image: opencti/platform:latest
        environment:
          - APP__PORT=8080
          - APP__ADMIN__EMAIL=p.tisserand@serma.com
          - APP__ADMIN__PASSWORD=L3soufledudr4g0n
          - APP__ADMIN__TOKEN=[myToken]
          - APP__LOGS_LEVEL=error
          - APP__LOGS=./logs
          - APP__REACTIVE=true
          - APP__COOKIE_SECURE=false
          - GRAKN__HOSTNAME=grakn
          - GRAKN__PORT=48555
          - GRAKN__TIMEOUT=30000
          - REDIS__HOSTNAME=redis
          - REDIS__PORT=6379
          - ELASTICSEARCH__URL=http://elasticsearch:9200
          - MINIO__ENDPOINT=minio
          - MINIO__PORT=9000
          - MINIO__USE_SSL=false
          - MINIO__ACCESS_KEY=${MINIO_ACCESS_KEY}
          - MINIO__SECRET_KEY=${MINIO_SECRET_KEY}
          - RABBITMQ__HOSTNAME=rabbitmq
          - RABBITMQ__PORT=5672
          - RABBITMQ__PORT_MANAGEMENT=15672
          - RABBITMQ__MANAGEMENT_SSL=false
          - RABBITMQ__USERNAME=${RABBITMQ_DEFAULT_USER}
          - RABBITMQ__PASSWORD=${RABBITMQ_DEFAULT_PASS}
          - PROVIDERS__LOCAL__STRATEGY=LocalStrategy
        ports:
          - "8080:8080"
        depends_on:
          - grakn
          - redis
          - elasticsearch
          - minio
          - rabbitmq
        restart: always
      worker:
        image: opencti/worker:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - WORKER_LOG_LEVEL=info
        depends_on:
          - opencti
        deploy:
          mode: replicated
          replicas: 3
        restart: always
      connector-export-file-stix:
        image: opencti/connector-export-file-stix:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - CONNECTOR_ID=b3d25d79-60bc-4764-9bdb-7ac1597249a2 # Valid UUDv4
          - CONNECTOR_TYPE=INTERNAL_EXPORT_FILE
          - CONNECTOR_NAME=ExportFileStix2
          - CONNECTOR_SCOPE=application/json
          - CONNECTOR_CONFIDENCE_LEVEL=3
          - CONNECTOR_LOG_LEVEL=info
        restart: always
      connector-import-file-stix:
        image: opencti/connector-import-file-stix:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - CONNECTOR_ID=b69a692c-09c5-44cc-96f7-8fd1d8d31e66 # Valid UUDv4
          - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
          - CONNECTOR_NAME=ImportFileStix2
          - CONNECTOR_SCOPE=application/json
          - CONNECTOR_CONFIDENCE_LEVEL=3
          - CONNECTOR_LOG_LEVEL=info
        restart: always
      connector-import-file-pdf-observables:
        image: opencti/connector-import-file-pdf-observables:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - CONNECTOR_ID=8aae5320-172a-4adb-aa44-805a556e8658 # Valid UUDv4
          - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
          - CONNECTOR_NAME=ImportFilePdfObservables
          - CONNECTOR_SCOPE=application/pdf
          - CONNECTOR_CONFIDENCE_LEVEL=3
          - CONNECTOR_LOG_LEVEL=info
          - PDF_OBSERVABLES_CREATE_INDICATOR=False
        restart: always    
      connector-opencti:
        image: opencti/connector-opencti:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - CONNECTOR_ID=8aae5320-172a-4adb-aa44-805a556e8658 # Valid UUDv4
          - CONNECTOR_TYPE=EXTERNAL_IMPORT
          - CONNECTOR_NAME=OpenCTI
          - CONNECTOR_SCOPE=identity,sector,region,country,city
          - CONNECTOR_CONFIDENCE_LEVEL=5
          - CONNECTOR_UPDATE_EXISTING_DATA=true
          - CONNECTOR_LOG_LEVEL=info
          - CONFIG_SECTORS_FILE_URL=https://raw.githubusercontent.com/OpenCTI-Platform/datasets/master/data/sectors.json
          - CONFIG_GEOGRAPHY_FILE_URL=https://raw.githubusercontent.com/OpenCTI-Platform/datasets/master/data/geography.json
          - CONFIG_INTERVAL=7 # Days
        restart: always
      connector-mitre:
        image: opencti/connector-mitre:latest
        environment:
          - OPENCTI_URL=http://127.0.0.1:8080
          - OPENCTI_TOKEN=[myToken]
          - CONNECTOR_ID=71d5d838-4a0c-40d3-99e8-9ba7ccba9de2 # Valid UUDv4
          - CONNECTOR_TYPE=EXTERNAL_IMPORT
          - CONNECTOR_NAME=MITRE ATT&CK
          - CONNECTOR_SCOPE=identity,attack-pattern,course-of-action,intrusion-set,malware,tool,report
          - CONNECTOR_CONFIDENCE_LEVEL=3
          - CONNECTOR_UPDATE_EXISTING_DATA=true
          - CONNECTOR_LOG_LEVEL=info
          - MITRE_ENTERPRISE_FILE_URL=https://raw.githubusercontent.com/mitre/cti/master/enterprise-attack/enterprise-attack.json
          - MITRE_PRE_ATTACK_FILE_URL=https://raw.githubusercontent.com/mitre/cti/master/pre-attack/pre-attack.json
          - MITRE_INTERVAL=7 # Days
        restart: always
    volumes:
      grakndata:
        driver: local
      esdata:
        driver: local
      s3data:
        driver: local
```

**Comment by darcosion** [comment]
Found my errors.

So : connector-import-file-pdf-observables and connector-opencti have a collision of ID because of an error in my conf (seeing https://github.com/OpenCTI-Platform/connectors/issues/54#issuecomment-595266518 ).

I've started my connectors in a specific docker-compose.yml or with just `python3 someconnector.py` and this isn't the good way. Connector should be integrate in the docker-compose.yml because he use `opencti:8080` and opencti is an alias for the Opencti API who have a virtual interface with a specific IP in the docker-compose (docker environnement).

Lastely, we had found than my `.env` wasn't configured, so many var contain errors.
